lib/ansible/modules/network/vyos/vyos_system.py

```diff
@@ -0,0 +1,214 @@
+      - sub1.example.com
+      - sub2.example.com
+"""
+
+from ansible.module_utils.local import LocalAnsibleModule
+from ansible.module_utils.vyos import get_config, load_config
+
+
+def spec_key_to_device_key(key):
+    device_key = key.replace('_', '-')
+
+    # domain-search is longer than just it's key
+    if device_key == 'domain-search':
+        device_key += ' domain'
+
+    return device_key
+
+
+def config_to_dict(module):
+    data = get_config(module)
+
+    config = {'domain_search': [], 'name_server': []}
+
+    for line in data.split('\n'):
+        if line.startswith('set system host-name'):
+            config['host_name'] = line[22:-1]
+        elif line.startswith('set system domain-name'):
+            config['domain_name'] = line[24:-1]
+        elif line.startswith('set system domain-search domain'):
+            config['domain_search'].append(line[33:-1])
+        elif line.startswith('set system name-server'):
+            config['name_server'].append(line[24:-1])
+
+    return config
+
+
+def spec_to_commands(want, have):
+    commands = []
+
+    state = want.pop('state')
+
+    # state='absent' by itself has special meaning
+    if state == 'absent' and all(v is None for v in want.values()):
+        # Clear everything
+        for key in have:
+            commands.append('delete system %s' % spec_key_to_device_key(key))
+
+    for key in want:
+        if want[key] is None:
+            continue
+
+        current = have.get(key)
+        proposed = want[key]
+        device_key = spec_key_to_device_key(key)
+
+        # These keys are lists which may need to  be reconciled with the device
+        if key in ['domain_search', 'name_server']:
+            if not proposed:
+                # Empty list was passed, delete all values
+                commands.append("delete system %s" % device_key)
+            for config in proposed:
+                if state == 'absent' and config in current:
+                    commands.append("delete system %s '%s'" % (device_key, config))
+                elif state == 'present' and config not in current:
+                    commands.append("set system %s '%s'" % (device_key, config))
+        else:
+            if state == 'absent' and current and proposed:
+                commands.append('delete system %s' % device_key)
+            elif state == 'present' and proposed and proposed != current:
+                commands.append("set system %s '%s'" % (device_key, proposed))
+
+    return commands
+
+
+def main():
+    argument_spec = dict(
+        host_name=dict(type='str'),
+        domain_name=dict(type='str'),
+        domain_search=dict(type='list'),
+        name_server=dict(type='list'),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+    )
+
+    module = LocalAnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[('domain_name', 'domain_search')],
+    )
+
```

Comment:
With no options, vyos_system with no options is equivalent to `{'state': 'present'}`, and effectively a no-op (other than collecting the current config). As `{'state': 'absent'}` by itself is a supported (and documented) option, the only possibility is `required_one_of(argument_spec.keys())`, which still lets the user do the same thing, just more explicitly.